### PR TITLE
Throw an error if result of ```NINT``` overflows its kind 

### PR DESCRIPTION
**Pull request by shivimathur**
Fixes : https://github.com/lfortran/lfortran/issues/4079

### CODE REVIEW

File: integration_tests/intrinsics_242.f90 [hunk 1]
```diff
@@ -0,0 +1,9 @@
+program intrinsics_242
+
```
Comment (certik) [inline integration_tests/intrinsics_242.f90:2]: Add:
```suggestion
    integer, parameter :: dp = kind(0.d0)
```

File: integration_tests/intrinsics_242.f90 [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+    print*, nint(1e12_8, 8)
+    if (nint(1e12_8, 8) /= 1000000000000_8) error stop
```
Comment (certik) [inline integration_tests/intrinsics_242.f90:4]: And use like this:
```suggestion
    if (nint(1e12_dp, dp) /= 1000000000000_dp) error stop
```
and so on.

File: src/libasr/pass/intrinsic_functions.h [hunk 3]
```diff
@@ -1992,9 +1992,23 @@ namespace Anint {
...
     static ASR::expr_t *eval_Nint(Allocator &al, const Location &loc,
-            ASR::ttype_t* arg_type, Vec<ASR::expr_t*> &args, diag::Diagnostics& /*diag*/) {
+            ASR::ttype_t* arg_type, Vec<ASR::expr_t*> &args, diag::Diagnostics& diag) {
+        int kind = ASRUtils::extract_kind_from_ttype_t(arg_type);
         double rv = ASR::down_cast<ASR::RealConstant_t>(args[0])->m_r;
         double near_integer = std::round(rv);
+
+        if (kind == 4) {
+            if (near_integer < std::numeric_limits<int32_t>::min() || near_integer > std::numeric_limits<int32_t>::max()) {
+                diag.semantic_error_label("Result of NINT overflows its kind(" + std::to_string(kind) + ")", {loc}, "");
```
Comment (certik) [inline src/libasr/pass/intrinsic_functions.h:2002]: Use lowercase for intrinsic functions:
```suggestion
                diag.semantic_error_label("Result of `nint` overflows its kind(" + std::to_string(kind) + ")", {loc}, "");
```

File: src/libasr/pass/intrinsic_functions.h [hunk 4]
```diff
@@ -1992,9 +1992,23 @@ namespace Anint {
...
+            }
+        } else if (kind == 8) {
+            if (near_integer < std::numeric_limits<int64_t>::min() || near_integer > std::numeric_limits<int64_t>::max()) {
+                diag.semantic_error_label("Result of NINT overflows its kind(" + std::to_string(kind) + ")", {loc}, "");
```
Comment (certik) [inline src/libasr/pass/intrinsic_functions.h:2006]:
```suggestion
                diag.semantic_error_label("Result of `nint` overflows its kind(" + std::to_string(kind) + ")", {loc}, "");
```

**Comment by certik** [comment]
Awesome, thanks for fixing that! I left some minor comments.

**Review by HarshitaKalani — APPROVED** [review]
Perfect, looks great to me! Thanks @shivimathur

**Comment by Pranavchiku** [comment]
@shivimathur would you like to clean commit history or a squash and merge would be fine?

**Comment by shivimathur** [comment]
> @shivimathur would you like to clean commit history or a squash and merge would be fine?

@Pranavchiku I think clean commit history would be good.

**Review by Pranavchiku — APPROVED** [review]
Perfect, thanks!

**Pull request merged**